dsl/src/main/scala/org/http4s/dsl/impl/Path.scala

```diff
@@ -326,6 +326,19 @@ abstract class OptionalQueryParamDecoderMatcher[T: QueryParamDecoder](name: Stri
       .toOption
 }
 
+abstract class DefaultQueryParamDecoderMatcher[T: QueryParamDecoder](name: String, default: T) {
+  def unapply(params: Map[String, collection.Seq[String]]): Option[T] =
```

Comment:
I think this should return `Some` rather than `Option`?

Reply:
Are you sure? What if `params.get(name)` is `None`?

Reply:
Isn't that what the default is for?

Sounds like this should have a doc to spell out the semantics :)

Reply:
Haha yes I derped there 😆 good point.

Reply:
Does it also use the default if there's a value but it fails to parse? (I'm having one of those days where my code-reading comprehension is extremely minimal.) I think that should be documented either way.

Reply:
But I remember looking at the tests and they seemed to do the right thing. So now I am confused.

Reply:
If it still compiles after changing the `Option` to `Some` then the behaviour is probably correct.

Reply:
It's `Option` because we still return `None` in the case that the parameter is present but with an incorrect format, eg as in one of the test cases. However I can change this behavior to return the default in that case if you prefer that

Reply:
Haha no problem, I'll add some docs

Reply:
I was going to say it should be changed, but then I saw the optional decoder returns `None` on the outer option in the case of a decoding failure. I disagree with that behaviour but I think for consistency we should do the same here. I'd like to reconsider that before 1.0 but I don't think we should hold this PR up on account of that.

Reply:
Interesting, look forward to hear your thoughts on that :)